File: app/src/main/java/yc/com/answer/index/contract/BookContract.java
package yc.com.answer.index.contract;

import yc.com.answer.index.model.bean.BookAnswerInfoWrapper;
import yc.com.base.IHide;
import yc.com.base.ILoading;
import yc.com.base.INoData;
import yc.com.base.INoNet;
import yc.com.base.IPresenter;
import yc.com.base.IView;

/**
 * Created by wanglin  on 2018/3/8 09:16.
 */

public interface BookContract {

    interface View extends IView, ILoading, INoData, INoNet, IHide {


        void showBookList(BookAnswerInfoWrapper data);

        void showEnd();
    }

    interface Presenter extends IPresenter {
    }
}

File: app/src/main/java/yc/com/homework/word/contract/BookContract.kt
package yc.com.homework.word.contract

import java.util.ArrayList

import yc.com.base.IDialog
import yc.com.base.IFinish
import yc.com.base.IHide
import yc.com.base.ILoading
import yc.com.base.IPresenter
import yc.com.base.IView
import yc.com.homework.word.model.domain.BookWordInfo

/**
 * Created by zhangkai on 2017/7/25.
 */

interface BookContract {
    interface View : IView, IDialog, IFinish, ILoading, IHide {
        fun showBookListData(list: ArrayList<BookWordInfo>?, isEdit: Boolean)

        fun deleteBookRefresh()
    }

    interface Presenter : IPresenter {
        fun bookList(currentPage: Int, pageCount: Int) //获取教材集合

        fun addBook(bookInfo: BookWordInfo?)

        fun deleteBook(bookInfo: BookWordInfo?)
    }
}
